### PR TITLE
Run Checkov via the Docker image

### PR DESCRIPTION
**Pull request by ZoogieZork**
## Description

Runs Checkov via the [Docker image](https://hub.docker.com/r/bridgecrew/checkov) instead of installed via pip/pipenv.

Additionally, cleaned up lint and typehint warnings, as well as improved the comments.

> [!NOTE]
> Checkov will be removed from the Pipfile as part of the next dependency update run.

## Motivation and Context

Checkov has a very large and picky list of dependencies, which been a major source of dependency conflicts.  By using the Docker image instead, this allows us to keep other dependencies updated while updating Checkov regularly.

## How Has This Been Tested?

Tested in non-prod environment.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change

## Checklist

- [x] My code follows conforms to the coding standards.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: backend/engine/plugins/checkov/main.py [hunk 1]
```diff
@@ -21,29 +34,60 @@ def main():
...
+
+    errors: list[str] = []
+
+    (temp_vol_name, temp_vol_mount) = str(args.engine_vars.get("temp_vol_name", "")).split(":")
+    if not temp_vol_name or not temp_vol_mount:
+        errors.append("Temporary volume not provided")
+
+    (working_src, working_mount) = str(args.engine_vars.get("working_mount", "")).split(":")
+    if not working_src or not working_mount:
+        errors.append("Working volume not provided")
```
Comment (g-marconet) [inline backend/engine/plugins/checkov/main.py:46]: How are `temp_vol_name` and `working_mount` defined? Do we control them?

Comment (ZoogieZork) [reply backend/engine/plugins/checkov/main.py:46]: These are set here: https://github.com/WarnerMedia/artemis/blob/zoogiezork/checkov-docker-2/backend/engine/utils/plugin.py#L167-L168

File: backend/engine/plugins/checkov/main.py [hunk 2]
```diff
@@ -144,10 +205,12 @@ def parse_checkov(checkov_output: list[dict], repo_path: str, config_dir: str, c
     return (findings, error)
 
 
-def get_config_dir(repo_path: str, config: dict) -> tuple[str, str]:
+def get_config_dir(config: dict) -> tuple[Path, Optional[str]]:
     """
     Determine if config directory should be from S3 or default (local)
     """
+    # TODO: Write to the temporary volume instead of the plugin source directory.
```
Comment (g-marconet) [inline backend/engine/plugins/checkov/main.py:212]: Nit: TODO possibly left in? Would this be better as an issue in an issue tracker? (also L222 below)

**Review by g-marconet — APPROVED** [review]
LGTM

File: backend/engine/plugins/checkov/main.py (hunk 2 above)
Comment (ZoogieZork) [reply backend/engine/plugins/checkov/main.py:212]: Yep, this is intentional -- it's a note for the next PR in this series (or if I don't get around to it, will be added to the backlog).

The existing implementation of this writes the retrieved configuration into the actual plugin source directory, which persists between runs.  This *technically* works, so doesn't block this PR, but is definitely not good and potentially cause issues in some corner cases.

As far as I can tell, we're not actually using this "feature" so it's a lower priority to fix.

**Pull request merged**